### ISSUE TITLE
Spyder 4 after Spyder 5 does not work

### ISSUE THREAD

**Issue by thistlillo** [spyder-ide/spyder#15263]
Sad story on Window 10.

I was using Spyder 4, happily, then decided to update to version 5.

Version 5 gave me some errors concerning spyder-kernels, but they looked like 'false alarms' because the application did work.

Then, all of a sudden, today Spyder 5 was no longer able to launch an interactive console. 

I uninstalled Spyder 5 and reinstalled the latest minor of version 4, that used to work fantastically on my machine.

Now, version 4 is not able to launch an interactive console because spyder-kernels is missing. I did re-install spyder-kernels both using `pip install` and `conda install`, but spyder 4 complains about the missing spyder-kernels (using python 3.8.8).

Basically, it worked fantastically, now after all these passage it does not work and I have again that incredible pain-in-the-neck app, named Kite, running on my machine (this is my fault: it is likely I hit the wrong key).

Please next time you decide to update your app, make more tests.

**Comment by steff456** [comment]
Hi @thistlillo,

The spyder-kernels error is a duplicate of #15091 and the fix will be in the 5.0.1 release that is coming soon.

Please remember to try not to mix pip with conda because you could damage your environment. In a new and clean environment install spyder 4 and it should work as before. 

```
conda create -n myenv python=3.x
conda activate myenv
conda install spyder=4
```

Also, Kite is an optional extension of Spyder and you can decline to install it and continue to use Spyder. Lastly, please take into account that every major release of the project will bring its bugs and you as an user can decide if upgrade as soon as the first version is released or wait a little longer for us to stabilize this branch.

Thanks for reporting!

**Comment by ccordoba12** [comment]
Although we understand your frustration, please be aware that this kind of messages are really not constructive and demotivate us a lot.

Now my answers about your message:

> Then, all of a sudden, today Spyder 5 was no longer able to launch an interactive console.

This is not helpful at all because there's no error message we can help you with. So we can't give you any support it about it.

> Now, version 4 is not able to launch an interactive console because spyder-kernels is missing

The right version of spyder-kernels for Spyder 4.2.5 is 1.10.2. Furthermore, the required spyder-kernels version for each Spyder version is clearly shown in the message we display in the IPython console.

So it's not enough to just run

    conda install spyder-kernels

For Spyder 4 you need to run

    conda install spyder-kernels=1.10.2

> I did re-install spyder-kernels both using pip install

Please don't use pip to install packages that are provided by Anaconda. If the problem with spyder-kernels persists, I'm afraid you'll have to reinstall Anaconda completely (please see [our video](https://www.youtube.com/watch?v=Ul79ihg41Rs) about it).

> Please next time you decide to update your app, make more tests.

Most of the things you mentioned are basically installation issues (except for bug with Spyder 5 not working), so no amount of tests could catch them because we always run our tests in a clean environment.

Furthermore, to solve those issues we publish Windows and macOS installers [here](https://github.com/spyder-ide/spyder/releases), so that people don't have to suffer through this painful update process. For next time, please try them instead of lashing out your frustration here.

**Comment by ccordoba12** [comment]
I reopened the issue in case you have constructive feedback, but if the tone of your messages continues to be the same, I'll consider this discussion to be finished.

**Comment by gb119** [comment]
I'm not going to shout at developers when I'm benefiting from their hard work! But perhaps in the spirit of a feature request for the next time there is a major version change would be to make the preferences stored in a non-overlapping way - I've been playing with Spyder 5.0.0 in one of my conda environments, and some things don't quite work - which is fine (I just wish I was better at finding issues before I report duplicates!), but the preferences for Spyder 4 in my other environments seem to have been trampled over by spyder 5.0.0 which makes going back to Spyder 4 a little more painful than would be ideal. Appreciate it's too late for this round, but perhaps for Spyder 5->6?

**Comment by ccordoba12** [comment]
> I'm not going to shout at developers when I'm benefiting from their hard work!

@gb119, thanks for your kind words.

> I've been playing with Spyder 5.0.0 in one of my conda environments

That's a very good way to test things. You can also try that for next time @thistlillo.

> Appreciate it's too late for this round, but perhaps for Spyder 5->6?

We did a huge refactoring behind the scenes in Spyder 5 to provide a more extensible API, which is the cause of that. We don't plan to do something like that in later releases though, but we'll take your suggestion it into account.

**Comment by duemme** [comment]
Hello, I'm having the same problem.
Spyder 5 tell me "An error ocurred while starting the kernel", "Your Python environment or installation doesn't have the spyder‑kernels module or the right version of it installed (>= 2.0.1 and < 2.1.0). Without this module is not possible for Spyder to create a console for you.

You can install it by running in a system terminal:

conda install spyder‑kernels

or

pip install spyder‑kernels"

I was using Spyder 4 with 3.9.1_8, with no problems.
Both Spyder and Python installed via Home Brew. The manually selected the interpreter look for the python bin installed by home brew.

Then I upgraded both python (3.9.1_8 -> 3.9.4) and Spyder (v. 5) via Home Brew and I get this error.
I selected the new interpreter but the problem is still there.

If run pip3 install spyder‑kernels

I get this error:
ERROR: Invalid requirement: 'spyder‑kernels'   

Do I need to wait for a new version of the software?

**Comment by ccordoba12** [comment]
Closing due to lack of response.
